Task 9.3/Task 9.3/Program.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task93
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Введите натуральное число");
+            int Number = int.Parse(Console.ReadLine());
+            int Number1 = Number;
+            int MaxNumber = 0;
+            int NumberOfDigits = 1;
+            do
+            {
+                MaxNumber = Math.Max(Number % 10, MaxNumber);
+                Number /= 10;
+                NumberOfDigits += 1;
+            }
+            while ((Number / 10) > 0);
+
+            int OrderNumber = NumberOfDigits;
+            int NumberOfDigits1 = NumberOfDigits;
+            for (int i = 0; i <= NumberOfDigits1; i++)
+            {
+                if ((Number1 % 10) == MaxNumber)
+                    OrderNumber = NumberOfDigits;
+                NumberOfDigits -= 1;
+                Number1 /= 10;
+            }
```

Comment:
1. Имена локальных переменных в C# принято начинать с маленькой буквы.
2.  Если учитывать приоритеты операций, можно не писать лишние скобки.
3. Отделяйте смысловые участи кода, например, циклы и операторы условия пустыми строками. Так код читается легче.
```suggestion
            int number = int.Parse(Console.ReadLine());
            int number1 = number;
            int maxNumber = 0;
            int numberOfDigits = 1;
            
            do
            {
                maxNumber = Math.Max(number % 10, maxNumber);
                number /= 10;
                numberOfDigits += 1;
            }
            while (number / 10 > 0);

            int orderNumber = numberOfDigits;
            int numberOfDigits1 = numberOfDigits;
            
            for (int i = 0; i <= NumberOfDigits1; i++)
            {
                if ((number1 % 10) == maxNumber)
                    orderNumber = numberOfDigits;
                    
                numberOfDigits -= 1;
                number1 /= 10;
            }
```
Если модифицировать алгоритм, то можно было бы обойтись без второго цикла. Подумайте как.